blob/azureblob/azureblob.go

```diff
@@ -161,48 +163,45 @@ func openURL(ctx context.Context, u *url.URL) (driver.Bucket, error) {
 	q := u.Query()
 	opts := &Options{}
 	ac := &AzureCreds{}
+	credPath := q["cred_path"]
+	if len(credPath) == 0 {
+		return nil, fmt.Errorf("azureblob: cred_path query parameter is required")
+	}
 
-	if credPath := q["cred_path"]; len(credPath) > 0 {
+	f, err := ioutil.ReadFile(credPath[0])
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(f, ac)
+	if err != nil {
+		return nil, err
+	}
 
-		f, err := ioutil.ReadFile(credPath[0])
+	if ac.AccountKey != "" {
+		serviceURL, err := ServiceURLFromAccountKey(ac.AccountName, ac.AccountKey)
 		if err != nil {
 			return nil, err
 		}
 
-		err = json.Unmarshal(f, ac)
+		credential, err := azblob.NewSharedKeyCredential(ac.AccountName, ac.AccountKey)
 		if err != nil {
 			return nil, err
 		}
 
-		if ac.AccountKey != "" {
-			serviceURL, err := ServiceURLFromAccountKey(ac.AccountName, ac.AccountKey)
-			if err != nil {
-				return nil, err
-			}
-
-			credential, err := azblob.NewSharedKeyCredential(ac.AccountName, ac.AccountKey)
-			if err != nil {
-				return nil, err
-			}
-
-			opts.Credential = *credential
-			return openBucket(ctx, serviceURL, u.Host, opts), err
-		} else {
-			serviceURL, err := ServiceURLFromSASToken(ac.AccountName, ac.SASToken)
-			if err != nil {
-				return nil, err
-			}
-
-			return openBucket(ctx, serviceURL, u.Host, opts), err
-		}
+		opts.Credential = *credential
+		return openBucket(ctx, serviceURL, u.Host, opts), err
+	}
 
-	} else {
-		return nil, fmt.Errorf("azureblob: fail, missing querystring cred_path")
+	serviceURL, err := ServiceURLFromSASToken(ac.AccountName, ac.SASToken)
+	if err != nil {
+		return nil, err
 	}
+	return openBucket(ctx, serviceURL, u.Host, opts), err
```

Comment:
nit: s/err/nil/ (more idiomatic to be explicit)

Reply:
Done.